posthog/temporal/data_imports/pipelines/postgres/postgres.py

```diff
@@ -439,14 +465,16 @@ def postgres_source(
 
             primary_keys = _get_primary_keys(cursor, schema, table_name)
             table = _get_table(cursor, schema, table_name)
-            chunk_size = _get_table_chunk_size(cursor, inner_query_with_limit, schema, table_name, logger)
+            chunk_size = _get_table_chunk_size(cursor, inner_query_with_limit, logger)
             rows_to_sync = _get_rows_to_sync(cursor, inner_query_without_limit, logger)
             partition_settings = _get_partition_settings(cursor, schema, table_name) if is_incremental else None
 
             # Fallback on checking for an `id` field on the table
             if primary_keys is None and "id" in table:
                 primary_keys = ["id"]
 
+            has_duplicate_primary_keys = _has_duplicate_primary_keys(cursor, schema, table_name, primary_keys)
```

Comment:
There is no reason to do this check if the primary keys are actually primary keys. This check only makes sense when using the fallback "id", so it should be within the `if` block above.